### PR TITLE
Add ANNIndex base and milvus db

### PR DESCRIPTION
**Pull request by shiyu22**
Signed-off-by: shiyu22 <shiyu.chen@zilliz.com>

### CODE REVIEW

File: towhee/__init__.py [hunk 1]
```diff
@@ -11,13 +11,13 @@
...
 # limitations under the License.
-import os
-from typing import List, Union
+from typing import Union
 from towhee.engine import register
 from towhee.engine.factory import ops, pipeline, DEFAULT_PIPELINES, _PipelineBuilder as Build
 from towhee.hparam import param_scope
 from towhee.hparam import HyperParameter as Document
 from towhee.functional import DataCollection, State, Entity
+from towhee.ann_index import Connector as connector
```
Comment (reiase) [inline towhee/__init__.py:20]: It would be better to use `connectors` as the module name.

File: towhee/ann_index/base.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        """
+        pass
+
+    def save(self):
+        """
+        Save the index.
+        """
+        pass
+
+    def __enter__(self):
```
Comment (reiase) [inline towhee/ann_index/base.py:62]: can we put the code of entering and exiting here?

File: towhee/functional/data_collection.py [hunk 3]
```diff
@@ -459,40 +460,66 @@ def range(*arg, **kws):
         """
         return DataCollection(range(*arg, **kws))
 
-    def batch(self, size, drop_tail=False) -> 'DataCollection':
+    def batch(self, size, drop_tail=False, raw=True):
```
Comment (reiase) [inline towhee/functional/data_collection.py:463]: can we just put the entities into a list here, and don't deal with the fields.

**Review by reiase — CHANGES_REQUESTED** [review]

File: towhee/functional/data_collection.py (hunk 3 above)
Comment (shiyu22) [reply towhee/functional/data_collection.py:463]: I have tried it but not work. If we yield a list of entities will got:
```bash
~/workspace/git/towhee/towhee/engine/factory.py in <listcomp>(.0)
     73     def __apply__(self, *arg, **kws):
     74         if isinstance(self._index[0], tuple): # Multi inputs.
---> 75             args = [getattr(arg[0], x) for x in self._index[0]]
     76         else: # Single input.
     77             args = [getattr(arg[0], self._index[0])]

AttributeError: 'list' object has no attribute 'path'
```
and if yield DataCollection directly, the args in op is like `(<towhee.hparam.hyperparameter.wrapper object at 0x7f95031df460>, <towhee.hparam.hyperparameter.wrapper object at 0x7f95031df520>)`.

And the example code is follows:
```python
import towhee
from towhee import connector

milvus_collection = connector.milvus(uri='tcp://localhost:19530/reverse_image_search')            
                    

with milvus_collection:
    (
        towhee.glob['path']('/Users/chenshiyu/workspace/data/pic/test/*.jpg')
              .image_decode['path', 'img']()
              .image_embedding.timm['img', 'vec'](model_name='resnet50')
              .runas_op['path', 'path'](func=lambda path: abs(hash(path)) % (10 ** 8))
              .batch(3)
              .ann_insert[('path', 'vec'), 'res'](ann_index=milvus_collection)
              .run()
    )
```

File: towhee/ann_index/base.py (hunk 2 above)
Comment (shiyu22) [reply towhee/ann_index/base.py:62]: What do you mean, such as `load` with `__exit__`? But I think each index will have different action with `__enter__` and `__exit__`.

File: towhee/__init__.py (hunk 1 above)
Comment (shiyu22) [reply towhee/__init__.py:20]: get it.

File: towhee/functional/data_collection.py (hunk 3 above)
Comment (reiase) [reply towhee/functional/data_collection.py:463]: @GuoRentong We run into this problem angin, the behavior of batch  has not been defined for entities. I'll open another issue for this problem.

**Review by reiase — CHANGES_REQUESTED** [review]

**Review by reiase — APPROVED** [review]

**Comment by shiyu22** [comment]
#1149

**Pull request merged**